src/Database/Query.php

```diff
@@ -402,7 +402,10 @@ public function with(CommonTableExpression|Closure $cte, bool $overwrite = false
             }
         }
 
-        $this->_parts['with'][] = $cte;
+        if ($cte) {
+            $this->_parts['with'][] = $cte;
+        }
```

Comment:
Not sure if a no-op like `with(null)` makes sense, but other query builder methods with override support allow similar no-ops.